app/services/publish/accredited_provider_updater.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Publish
+  class AccreditedProviderUpdater
+    attr_reader :provider, :new_accredited_provider, :recruitment_cycle
+
+    def initialize(provider_code:, recruitment_cycle_year:, new_accredited_provider_code:)
+      @recruitment_cycle = RecruitmentCycle.find_by!(year: recruitment_cycle_year)
+      @provider = Provider.find_by!(provider_code:, recruitment_cycle: @recruitment_cycle)
+      @new_accredited_provider = Provider.find_by!(provider_code: new_accredited_provider_code, recruitment_cycle: @recruitment_cycle)
+    end
+
+    def update_provider_and_courses
+      update_provider && update_courses
```

Comment:
Does it matter if one returns true and the other false and when returns false we don't know what fail? 
(if was the first or second condition)

Reply:
Both the methods execute SQL directly so unlikely to return false and more likely to fail/error. 

We're under a lot of pressure to update these accredited providers beofre the end of the month so that Register data is accurate.